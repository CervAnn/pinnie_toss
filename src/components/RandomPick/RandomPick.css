.RandomPick {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  flex-flow: row wrap;
  width: 100%;
  padding: 40px;
}

#randomPick-text {
  text-align: center;
  font-size: 35px;
  width: 100%;
  margin-bottom: 10px;
  color: white;
  text-shadow: -1px -1px 0 #193d7b, 1px -1px 0 #193d7b, -1px 1px 0 #193d7b,
    1px 1px 0 #193d7b;
}
