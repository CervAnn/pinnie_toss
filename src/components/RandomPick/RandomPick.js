import React from 'react'
import { connect } from 'react-redux'
import NFLPick from '../NFLPick/NFLPick'
import NCAAPick from '../NCAAPick/NCAAPick'
import './RandomPick.css'

export const RandomPick = ({randomTeam}) => {
  return (
    <section className="RandomPick">
      <h2>Congratulations! You are now a proud fan of the:</h2>
      {randomTeam.strLeague === "NFL" ? <NFLPick /> : <NCAAPick />}
    </section>
  )
}

export const mapStateToProps = store => ({
  teams: store.teams,
  randomTeam: store.randomTeam
})


export default connect(mapStateToProps)(RandomPick);