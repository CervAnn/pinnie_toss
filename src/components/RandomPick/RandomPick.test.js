import React from "react";
import { shallow } from "enzyme";
import { RandomPick, mapStateToProps } from "./RandomPick";

describe("RandomPick", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <RandomPick
        league = {{}}
      />
    );
  });

  it("should render without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("mapStateToProps", () => {
    it("should return an object of league and its contents", () => {
      let mockStore = {
        teams: [
          {
            idTeam: "134920",
            strLeague: "NFL",
            strTeam: "New England Patriots",
          },
          {
            idTeam: "134922",
            strLeague: "NFL",
            strTeam: "Baltimore Ravens",
          }
        ],
        randomTeam: {
          idTeam: "136915",
          idLeague: "4479",
          strTeam: "Michigan",
          strAlternate: "Wolverines",
        },
        league: {
          leagueName: "NCAA",
          leagueId: 4479,
          leagueDivision: "FBS"
        }
      };

      let expected = {
          league : {
          leagueName: "NCAA",
          leagueId: 4479,
          leagueDivision: "FBS"
        }
      }

      expect(mapStateToProps(mockStore)).toEqual(expected);
    });
  });
});
