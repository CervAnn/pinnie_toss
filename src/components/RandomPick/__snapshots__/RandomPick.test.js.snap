// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomPick should render without crashing 1`] = `
<section
  className="RandomPick"
>
  <h2
    id="randomPick-text"
  >
    Congratulations! You are now a proud fan of the:
  </h2>
  <Connect(NFLPick) />
</section>
`;
